datafusion/physical-expr/src/analysis.rs

```diff
@@ -64,11 +65,32 @@ impl AnalysisContext {
     ) -> Self {
         let mut column_boundaries = vec![];
         for (idx, stats) in statistics.iter().enumerate() {
-            column_boundaries.push(ExprBoundaries::from_column(
-                stats,
-                input_schema.fields()[idx].name().clone(),
-                idx,
-            ));
+            let field: &Arc<Field> = &input_schema.fields()[idx];
+            if let Ok(inf_field) = ScalarValue::try_from(field.data_type()) {
```

Comment:
I think it would also be ok here to return an internal error if we can't create a `ScalarValue` (not much else in DataFusion is going to work). And it would make clear we aren't silently ignoring errors.